### DISCUSSION TITLE
Separation of constraints and indices from the fields

### DISCUSSION THREAD

**Discussion by Attackwave** [TiagoSilvaPereira/vemto-issues#258] [Ideas & Suggestions]
Hi,

as a database developer, I create constraints and indices separately. Is the proposal a way for Vemto to do just that? A primary key with 1-n fields and several indices with 1-n fields each. The constraints and indices can be named if laravel can handle this.

```
$table->primary('id', 'pk_test');
$table->primary(['id', 'id2'], 'pk_test');

$table->unique('a', 'u1_test');
$table->unique(['a', 'b'], 'u1_test');

$table->index('x', 'i1_test');
$table->index(['x', 'y'], 'i1_test');
```

As I have read, Laravel only knows Primary Key, Foreign Keys and Indices.
![image](https://user-images.githubusercontent.com/51136146/144133176-a7efe6d9-da7d-4ca5-b61c-b2bb3bc193f9.png)

Separate JSON Configs for migration template

**Comment by TiagoSilvaPereira** [comment]
@Attackwave thank you for your suggestion
